### PR TITLE
[DVT-85] add goreleaser to erigon github actions

### PR DESCRIPTION
**Pull request by rebelArtists**
to streamline the release process of new maticnetwork/erigon code changes, we've added a github actions automated workflow with goReleaser to support a wide variety of operating systems and linux distributions

linear ticket: https://linear.app/matic/issue/DVT-85/erigon-ci-flow

leveraging similar goReleaser structure already in place for bor, as seen [here](https://github.com/maticnetwork/bor/blob/develop/.goreleaser.yml)

the release process is triggered upon pushing a semantic-versioned tag, for example: 
`git tag v0.0.3 091ae61`
`git push origin dan/goreleaser_erigon  tag v0.0.3`

successful goReleaser github action run logs here: https://github.com/maticnetwork/erigon/runs/7202381340?check_suite_focus=true

further confirmations:
- initial dockerhub release with this system successful, see: https://hub.docker.com/r/0xpolygon/erigon/tags
- pulling down the dockerhub arm64 image locally, erigon is able to run successful, with following command:
`docker run 0xpolygon/erigon:0.0.2-alpha-arm64  --chain=mainnet \
    --datadir /var/lib/erigon/data`
image
- binaries also properly uploaded to github releases, see: https://github.com/maticnetwork/erigon/releases

### CODE REVIEW

File: builder/files/erigon.service [hunk 1]
```diff
@@ -0,0 +1,34 @@
+[Unit]
+  Description=erigon
+  StartLimitIntervalSec=500
+  StartLimitBurst=5
+
+[Service]
+  Restart=on-failure
+  RestartSec=5s
+  ExecStart=/usr/local/bin/erigon \
```
Comment (praetoriansentry) [inline builder/files/erigon.service:9]: Do you have to specify the `bindir` for nfpm or does it default to this location

Comment (rebelArtists) [reply builder/files/erigon.service:9]: manually copy erigon binary to this dir in dockerfile.release here: https://github.com/maticnetwork/erigon/pull/3/files#diff-701366a9e47be6026c2b3a9cc9484a02f319b9b8469b28e907ac1237fe2456caR5

**Comment by rebelArtists** [comment]
> We need to do some testing of the released artifacts. I think it's going to take some work to support operating systems that aren't using systemd or situations where people want to customize which network they're connecting to. I think for now, it's possible just to remove the service file and ship the binary.

agreed, seems an opportunity for us to setup up an automatic testing system for the releases on various linux distributions.. for now, i've removed the nfpms section  / service file and i can file a ticket for this be a follow-on task

**Review by vcastellm — COMMENTED** [review]
Late LGTM but no deb/rpm packages?

**Pull request merged**